src/main/java/org/jmhsrobotics/frc2024/subsystems/drive/commands/IntakeCommand.java

```diff
@@ -1,46 +1,91 @@
-    }
-
-    @Override
-    public void initialize() {
-        // TODO Auto-generated method stub
-        this.timer.reset();
-        this.timer.start();
-    }
-
-    @Override
-    public void execute() {
-        // TODO Auto-generated method stub
-        SmartDashboard.putNumber("Intakecommand/intakespeed", this.speed);
-    }
-
-    @Override
-    public boolean isFinished() {
-        // TODO Auto-generated method stub
-        return timer.hasElapsed(5);
-    }
-
-    @Override
-    public void end(boolean interrupted) {
-        // TODO Auto-generated method stub
-        SmartDashboard.putNumber("Intakecommand/intakespeed", Double.NaN);
-    }
-}
\ No newline at end of file
+	protected CANSparkMax innerIntakeMotor;
+	protected Solenoid intakeArmsVirtualSolenoidA;
+	protected Solenoid intakeArmsVirtualSolenoidB;
+	protected CANSparkMax intakeArmsMotor;
+	protected DigitalInput intakeSwitch;
+
+	// protected boolean isClimbing;
+	public IntakeCommand(DriveSubsystem driveSubsystem, int i) {
+		innerIntakeMotor = new CANSparkMax(IntakeConstants.innerIntakeMotor, MotorType.kBrushless);
+		innerIntakeMotor.restoreFactoryDefaults();
+		innerIntakeMotor.setInverted(true);
+		innerIntakeMotor.setIdleMode(IdleMode.kBrake);
+
+		innerIntakeMotor.setSmartCurrentLimit(35);
+
+		intakeArmsVirtualSolenoidA = new Solenoid(PneumaticsModuleType.CTREPCM, 3);
+		intakeArmsVirtualSolenoidB = new Solenoid(PneumaticsModuleType.CTREPCM, 4);
+		intakeArmsVirtualSolenoidA.set(false);
+		intakeArmsVirtualSolenoidB.set(false);
+
+		intakeArmsMotor = new CANSparkMax(IntakeConstants.intakeArmsMotor, MotorType.kBrushless);
+		intakeArmsMotor.restoreFactoryDefaults();
+		intakeArmsMotor.setInverted(true);
+
+		intakeArmsMotor.setSmartCurrentLimit(35);
+
+		intakeSwitch = new DigitalInput(IntakeConstants.intakeSwitch);
+	}
+	public void enableInnerIntakeMotor() {
+		innerIntakeMotor.set(1);
+	}
+	public void reverseInnerIntakeMotor() {
+		innerIntakeMotor.set(-1);
+	}
+	public void disableInnerIntakeMotor() {
+		innerIntakeMotor.set(0);
+	}
+	public void setInnerIntakeMotor(double speed) {
+		this.innerIntakeMotor.set(speed);
+	}
+
+	// implementing 836's idea
+	public void extendIntakeArms() {
+		intakeArmsVirtualSolenoidA.set(true);
+		intakeArmsVirtualSolenoidB.set(true);
+	}
+
+	// implementing 836's idea
+	public void retractIntakeArms() {
+		intakeArmsVirtualSolenoidA.set(false);
+		intakeArmsVirtualSolenoidB.set(false);
+	}
+
+	// implementing 836's idea
+	public void floatIntakeArms() {
+		intakeArmsVirtualSolenoidA.set(false);
+		intakeArmsVirtualSolenoidB.set(true);
+	}
+
+	public void enableIntakeArmsMotor() {
+		intakeArmsMotor.set(-0.6);
+	}
+	public void disableIntakeArmsMotor() {
+		intakeArmsMotor.set(0);
+	}
+	public void reverseIntakeArmsMotor() {
+		intakeArmsMotor.set(0.6);
+	}
+	public double extendedIntakeArmsSpeed() {
+		return intakeArmsMotor.get();
+	}
+
+	public boolean getIntakeSwitch() {
+		return intakeSwitch.get();
+	}
```

Comment:
Rename the method here to be more externally accessible. IE isFull or hasNote or something similar that lets you know the parity of the state you are getting. IE true means full false means empty. Right not it is not clear is intake switch of true means a game object exists or not.